feat: add function to manually check rate limit (#392)
fastify/fastify-rate-limit: README.md

})
```

### Manual Rate Limit

A custom limiter function can be created with `fastify.createRateLimit()`, which is handy when needing to integrate with
technologies like [GraphQL](https://graphql.org/) or [tRPC](https://trpc.io/). This function uses the global [options](#options) set
during plugin registration, but you can override options such as `store`, `skipOnError`, `max`, `timeWindow`,
`allowList`, `keyGenerator`, and `ban`.

Example usage:

```js
import Fastify from 'fastify'

const fastify = Fastify()

// register with global options
await fastify.register(import('@fastify/rate-limit'), {

>   max: 100,

Reviewer: without it, routes will have 2 rate limit hooks

```suggestion
  global : false,
  max: 100,
```

  timeWindow: '1 minute'
})

>  // checkRateLimit will use the global options provided above when called

Reviewer: ```suggestion
// checkRateLimit will use the global options provided above when called
```

})
```

### Manual Rate Limit

> In some situations, the default behavior of this plugin may not be sufficient and you want to implement your own rate limit strategy.
> This might be the case as well if you want to integrate it with other technologies such as [GraphQL](https://graphql.org/) or [tRPC](https://trpc.io/).
> You can create your own limiter function with `fastify.createRateLimit()`.
> By default, this limiter function uses the global [options](#options) you defined during plugin registration.
> You can override some or all of your global options by passing them to `createRateLimit()`.
> The following options can be overridden: `store`, `skipOnError`, `max`, `timeWindow`, `allowList`, `keyGenerator`, and `ban`.
> 
> The next example demonstrates the usage of a custom rate limiter:

Reviewer 2: ```suggestion
A custom limiter function can be created with `fastify.createRateLimit()`, which is handy when needing to integrate with
technologies like [GraphQL](https://graphql.org/) or [tRPC](https://trpc.io/). This function uses the global [options](#options) set
during plugin registration, but you can override options such as `store`, `skipOnError`, `max`, `timeWindow`,
`allowList`, `keyGenerator`, and `ban`.

Example usage:
```

const fastify = Fastify()

// register with global options
await fastify.register(import('@fastify/rate-limit'), {
  max: 100,
  timeWindow: '1 minute'
})

 // checkRateLimit will use the global options provided above when called
const checkRateLimit = fastify.createRateLimit();

fastify.get("/", async (request, reply) => {
  // manually check the rate limit (using global options)
  const limit = await checkRateLimit(request);

  if(!limit.isAllowed && limit.isExceeded) {
    return reply.code(429).send("Limit exceeded");
  }

  return reply.send("Hello world");
});

// override global max option
const checkCustomRateLimit = fastify.createRateLimit({ max: 100 }); 

fastify.get("/custom", async (request, reply) => {
  // manually check the rate limit (using global options and overridden max option)
  const limit = await checkCustomRateLimit(request);

  // manually handle limit exceedance
  if(!limit.isAllowed && limit.isExceeded) {
    return reply.code(429).send("Limit exceeded");
  }

  return reply.send("Hello world");
});
```

A custom limiter function created with `fastify.createRateLimit()` only requires a `FastifyRequest` as the first parameter:

> ```ts

Reviewer 2: ```suggestion
```js
```
